Automattic/jetpack
# Migrate Add migration for new storage structure

Fixes #

## Proposed changes:

#### Description
This PR adds the necessary migration to move all files that were inside the `zbscrm-store` folder with a flat struture to the new `jpcrm-storage` folder that uses a hierarchical structure.


### Other information:


## Jetpack product discussion


Migrated from https://github.com/Automattic/zero-bs-crm/pull/2794
## Does this pull request change what data or activity we track or use?


No
## Testing instructions:






#### Testing instructions:

1. Add a custom file upload box in the `Custom Field` settings page (`wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=customfields`)
2. Checkout our v5.3.0 tag
3.  Add at least one file to: contacts, contacts using the custom upload box, invoices, quotes, company (b2b)
4. All files should be inside the `uploads/zbscrm-store` folder in a flat structure
5. Checkout `fix/2623/new_folder_structure_migration` and visit any CRM page to trigger the new migration
6. Verify in `wp-admin/admin.php?page=zerobscrm-systemstatus&tab=status` if the migration has run succesfully

All files should have been moved to the new folder (`wp-content/jpcrm-storage`) with a hierarchical structure.
All links should be working as expected.
